lib/ansible/module_utils/common/json.py

```diff
@@ -49,6 +49,15 @@ def __init__(self, preprocess_unsafe=False, vault_to_text=False, **kwargs):
         self._vault_to_text = vault_to_text
         super(AnsibleJSONEncoder, self).__init__(**kwargs)
 
+    def encode(self, o):
+        if getattr(o, '__ENCRYPTED__', False):
+            # vault object
+            if self._vault_to_text:
+                o = to_text(o, errors='surrogate_or_strict')
+            else:
+                o = {'__ansible_vault': to_text(o._ciphertext, errors='surrogate_or_strict', nonstring='strict')}
+        return json.JSONEncoder.encode(self, o)
+
     # NOTE: ALWAYS inform AWS/Tower when new items get added as they consume them downstream via a callback
     def default(self, o):
         if getattr(o, '__ENCRYPTED__', False):
```

Comment:
do we still need this entry?

Reply:
well, if I leave this entry out the json encoder just encodes it as a string without consulting the `default` method, which breaks everything

Reply:
i mean the entry in the default that check encrypted, now that it is in encode, it should never get there.

Reply:
oh sorry, jup